### DISCUSSION TITLE
Question about `--upload-delay` mode and `--cache-size`

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2841] [Q&A] [answered]
Suppose this mount:

`juicefs mount --writeback --upload-delay 2h --cache-size 2 ...`

1) Does the `--cache-size` impacts the `--writeback` mode so only 2Gb of `rawstaging` + normal cache (total) is allowed?
2) Or will the `rawstaging` folder **grown as much as necessary** far beyond 2Gb (configured size of `--cache-size`) as long as there is local disk space available, implied by the 2h upload mode (so if 10Gb of data is writen, it will be stored 100% locally and uploaded to the object storage 2h later)?

In other words, is `--cache-size` for cache-related operations only (GET/HEAD), or is it also used for `--writeback --upload-delay` options (PUT/DELETE)?

Thanks (and maybe document it also?).

**Comment by davies** [comment, marked as answer]
1. The size of rawstaging is not limited by `--cache-size`, which is used for read cache in `raw/`.

2. Yes.
